loki/backend/pygen.py

```diff
@@ -75,6 +77,21 @@ def map_array_subscript(self, expr, enclosing_prec, *args, **kwargs):
     def map_string_concat(self, expr, enclosing_prec, *args, **kwargs):
         return ' + '.join(self.rec(c, enclosing_prec, *args, **kwargs) for c in expr.children)
 
+    def map_inline_call(self, expr, enclosing_prec, *args, **kwargs):
+        arguments = ', '.join(self.rec(p, PREC_NONE, *args, **kwargs) for p in expr.parameters)
+
+        if expr.kw_parameters:
+            arguments += ', ' + ', '.join(
+                f'{self.rec(k, PREC_NONE, *args, **kwargs)}={self.rec(v, PREC_NONE, *args, **kwargs)}'
+                for k, v in expr.kw_parameters.items()
+            )
+
+        f = self.rec(expr.function, PREC_NONE, *args, **kwargs)
+        return self.format(f'{str(f).lower()}({arguments})')
```

Comment:
Should the `.lower()` not be something that is done in the Transformation rather than the backend? This effectively prohibits the use of function calls to routines with upper case function names.

Reply:
Yes, indeed! Good catch, these were actually masking some severe issues in our downcasing utility transform. I've just pushed two commits that remove these two short-cuts, and instead fix the issue in the F2P transformation, as suggested. :grin: